pkg/cli/cmds/certs.go

```diff
@@ -0,0 +1,53 @@
+package cmds
+
+import (
+	"github.com/rancher/k3s/pkg/version"
+	"github.com/urfave/cli"
+)
+
+const CertCommand = "certificate"
+
+var (
+	ComponentList    cli.StringSlice
+	CertCommandFlags = []cli.Flag{
+		DebugFlag,
+		ConfigFlag,
+		LogFile,
+		AlsoLogToStderr,
+		cli.StringFlag{
+			Name:        "data-dir,d",
+			Usage:       "(data) Folder to hold state default /var/lib/rancher/" + version.Program + " or ${HOME}/.rancher/" + version.Program + " if not root",
+			Destination: &ServerConfig.DataDir,
+		},
+		cli.StringSliceFlag{
+			Name:  "service,s",
+			Usage: "List of services to rotate certificates for. Options include (admin, api-server, controller-manager, scheduler, " + version.Program + "-controller, cloud-controller, etcd, auth-proxy, kubelet, kube-proxy)",
```

Comment:
Is there a default, or does the user need to manually specify every service (the code calls them components, but the help calls them services) that the cert should be rotated for?

* do we want to be consistent about calling these services, or components? Are they all services?
* which of these are valid on agents, and which are valid on servers? What happens if you try to rotate server component certificates on an agent?

Reply:
- you are right about the services/components ambiguity, I can just use services, to be more like rke1
- for the second point, in my mind, the tool wasnt intended to be surgical, basically if the user ran on agent or server it will rotate whatever it can find, thats because I didnt want to hardcode bunch of paths, but in any case I think I can change that behavior to make the subcommand detect the server or the agent from the directory structure and then fill up the serviceList or componentList with the list of certs that can be rotated that will also address this https://github.com/k3s-io/k3s/pull/4495#discussion_r755510507